src/iminuit/cost.py

```diff
@@ -116,7 +116,7 @@ class documentation for details.
 
 def _safe_log(x):
     # guard against x = 0
-    return np.log(x + 1e-323)
+    return np.log(np.maximum(np.finfo(x.dtype).smallest_normal, x))
```

Comment:
It's probably faster if `np.finfo(x.dtype).smallest_normal` is placed in a separate variable so that it doesn't need to be evaluated every time. You can find my current patch [here](https://salsa.debian.org/science-team/iminuit/-/blob/95b55b5e26a8796f65185cd89167903f6451720d/debian/patches/Fix-safe-log-precision.patch) (includes a pytest adjustment).

Reply:
In that case, we cannot select the subnormal that matches the dtype of x, but the current code does not work with numba anyway, so probably that's the solution to go for.

Reply:
I went for your solution, could you check again? Now Numba works again.